libmachine/provision/alpine.go

```diff
@@ -136,6 +136,11 @@ func (provisioner *AlpineProvisioner) Provision(swarmOptions swarm.Options, auth
 		}
 	}
 
+	log.Debug("Add Community repo")
+	if _, err := provisioner.SSHCommand("if ! apk info docker >/dev/null; then ver=$(awk '{split($1,a,\".\"); print a[1]\".\"a[2]}' /etc/alpine-release); echo \"http://dl-cdn.alpinelinux.org/alpine/v$ver/community\" >> /etc/apk/repositories; apk update; fi"); err != nil {
```

Comment:
I feel like this could be better for detecting whether community is enabled. Strictly speaking docker could be installed but community not although I'm not entirely sure in this case that it matters

Reply:
Would you prefer `if ! which docker >/dev/null && ! apk info docker >/dev/null; ...`?

Reply:
I'll think about it. It's probably fine for now

Reply:
Ok I've added the check in the last commit do you want me to remove it?

Reply:
Just to summarise what this script does: 
If the `docker` command is not found and `apk` can't find docker in any installed repositories via `apk info docker` we continue to install the community repo. Next we retrieve the major and minor version from `/etc/alpine-release`. The official mirror (CDN) for the appropriate version is appended to the `/etc/apk/repositories` file. Finally `apk update` is used to refresh the local repository database.

I think that this is pretty safe but if you have any ideas to improve this I'm open to improvements.

Reply:
I just spent some time putting this together, tell me what you think:

```
grep '^[[:blank:]]*[^#].*community' /etc/apk/repositories || sed
 -i '/^\s*[^#].*main/{p;s/main/community/;}' /etc/apk/repositories
```

This checks for the existence of an enabled community repo. If it's not present, it copies the `main` repository and `s/main/community/`.

From what I have tested this works in pretty much every case

Reply:
Yes I think this is much better. Would you like to make the commit so you get credit for the code?

The only way i can think of where this would not work in the highly unlikely case where when 'main' is not installed. But again I don't think that will happen. Maybe a user tries to run this script from the alpine.iso directly?

Reply:
Oh, hum. I didn't consider that case. Ideally this would be more complex to account for that too. Maybe I should add the extra check in if the `main` repo doesn't exist either. I'll make a PR to your fork in a minute

Reply:
I'll have to commit this later, but here is the revised statement:

```
if ! grep -q '^[[:blank:]]*[^#].*community$' /etc/apk/repositories; then grep -q '^[[:blank:]]*[^#].*main$' /etc/apk/repositories || echo "http://dl-cdn
.alpinelinux.org/alpine/v$(cut -d. -f1-2 /etc/alpine-release)/main" >> /etc/apk/repositories; sed -i '/^\s*[^#].*main/{p;s/main/community/;}' /etc/apk/repositories; fi; cat /etc/apk/repositories
```